src/main/java/com/redhat/rhjmc/containerjfr/platform/internal/KubeEnvPlatformClient.java

```diff
@@ -57,26 +58,33 @@
     private static final Pattern SERVICE_ENV_PATTERN =
             Pattern.compile("([\\S]+)_PORT_([\\d]+)_TCP_ADDR");
     private final Environment env;
+    private final Logger logger;
 
-    KubeEnvPlatformClient(Environment env) {
+    KubeEnvPlatformClient(Environment env, Logger logger) {
         this.env = env;
+        this.logger = logger;
     }
 
     @Override
-    public List<ServiceRef> listDiscoverableServices() {
+    public List<ServiceRef> listDiscoverableServices(Logger logger) {
         return env.getEnv().entrySet().stream()
-                .map(KubeEnvPlatformClient::envToServiceRef)
+                .map(this::envToServiceRef)
                 .filter(Objects::nonNull)
                 .collect(Collectors.toList());
     }
 
-    private static ServiceRef envToServiceRef(Map.Entry<String, String> entry) {
+    private ServiceRef envToServiceRef(Map.Entry<String, String> entry) {
         Matcher matcher = SERVICE_ENV_PATTERN.matcher(entry.getKey());
         if (!matcher.matches()) {
             return null;
         }
         String alias = matcher.group(1).toLowerCase();
         int port = Integer.parseInt(matcher.group(2));
-        return new ServiceRef(entry.getValue(), alias, port);
+        try {
+            return new ServiceRef(entry.getValue(), alias, port);
+        } catch (Exception e) {
+            logger.info(e);
```

Comment:
This goes for all of these new logger statements - maybe a Warning level message would be more appropriate?

Reply:
fixed.